Foxtrot/Foxtrot/Checker/VisibilityHelper.cs

```diff
@@ -37,6 +41,22 @@ private void ReInitialize(Method asThisMember)
             this.AsThisMember = asThisMember;
         }
 
+        private static bool IsAttribute(TypeNode declaringType)
+        {
+            if (declaringType == null || declaringType.BaseType == null)
+            {
+                return false;
+            }
+
+            if (declaringType == systemAttributeType.Value || declaringType.BaseType == systemAttributeType.Value)
+            {
+                return true;
+            }
+            
+
+            return IsAttribute(declaringType.BaseType);
+        }
+
```

Comment:
Current code checks declaringType.BaseType twice on recurrence.
Loop version should be simpler:
```C#
private static bool IsAttribute(TypeNode declaringType)
{
    TypeNode currentType = declaringType;
    while (currentType != null)
	{
	    if (currentType == systemAttributeType.Value)
		{
		    return true;
		}
		
		currentType = declaringType.BaseType;
	}
	
	return false;
}
```